great_expectations/data_context/types/data_context_variables.py

```diff
@@ -0,0 +1,154 @@
+    evaluation_parameter_store_name: Optional[str] = None
+    checkpoint_store_name: Optional[str] = None
+    profiler_store_name: Optional[str] = None
+    plugins_directory: Optional[str] = None
+    validation_operators: Optional[dict] = None
+    stores: Optional[dict] = None
+    data_docs_sites: Optional[dict] = None
+    notebooks: Optional[dict] = None
+    config_variables_file_path: Optional[str] = None
+    anonymous_usage_statistics: Optional[dict] = None
+    store_backend_defaults: Optional[dict] = None
+    concurrency: Optional[dict] = None
+    progress_bars: Optional[dict] = None
+
+    def __post_init__(self) -> None:
+        self._store = None
+
+    @property
+    def store(self) -> "VariablesStore":  # noqa: F821
+        if self._store is None:
+            self._store = self._init_store()
+        return self._store
+
+    @abstractmethod
+    def _init_store(self) -> "VariablesStore":  # noqa: F821
+        raise NotImplementedError
+
+    @staticmethod
+    def _get_key(attr: VariablesSchema) -> StringKey:
+        key: StringKey = StringKey(key=attr.value)
+        return key
+
+    def _set(self, attr: VariablesSchema, value: Any) -> None:
+        setattr(self, attr.value, value)
+        key: StringKey = DataContextVariables._get_key(attr)
+        self.store.set(key=key, value=value)
+
+    def _get(self, attr: VariablesSchema) -> Any:
+        key: StringKey = DataContextVariables._get_key(attr)
+        val: Any = self.store.get(key=key)
+        return val
+
+    def set_config_version(self, config_version: float) -> None:
+        """
+        TBD
+        """
+        self._set(VariablesSchema.CONFIG_VERSION, config_version)
+
+    def get_config_version(self) -> Optional[float]:
+        """
+        TBD
+        """
+        return self._get(VariablesSchema.CONFIG_VERSION)
+
+
+@dataclass
+class EphemeralDataContextVariables(DataContextVariables):
+    def _init_store(self) -> "VariablesStore":  # noqa: F821
+        from great_expectations.data_context.store.variables_store import VariablesStore
+
+        store: VariablesStore = VariablesStore(
+            store_name="ephemeral_data_context_variables_store",
+            store_backend=None,  # Defaults to InMemoryStoreBackend
+            runtime_environment=None,
+        )
+        return store
+
+
+@dataclass
+class FileDataContextVariables(DataContextVariables):
+    def _init_store(self) -> "VariablesStore":  # noqa: F821
+        from great_expectations.data_context.store.variables_store import VariablesStore
+
+        store_backend: dict = {"class_name": "InlineStoreBackend"}  # TBD
+        store: VariablesStore = VariablesStore(
+            store_name="file_data_context_variables_store",
+            store_backend=store_backend,
+            runtime_environment=None,
+        )
+        return store
+
+
+@dataclass
+class CloudDataContextVariables(DataContextVariables):
+    def __post_init__(
```

Comment:
TIL `__post_init__` ❤️

Reply:
necessary here because I'm trying to set an arg that doesn't have a default value (the Python interpreter yells about not having default values at the end of the signature). It's also a nice way to hack the magic of dataclasses!